frontend/packages/operator-lifecycle-manager/src/components/operand.tsx

```diff
@@ -64,12 +64,19 @@ const csvName = () =>
     );
 
 const getActions = (selectedObj: any) => {
+  let group = '';
+
+  if (selectedObj.apiVersion.includes('/')) {
```

Comment:
We shouldn't be doing any parsing of the `apiVersion` string here. All of that logic is already handled elsewhere.

Reply:
In order to verify the `group`, I have passed the `kindObj<k8sKind>` from [here](https://github.com/openshift/console/pull/4352/files#diff-bc19a445cd55d99705e404fae2c786edR598) which does not have `apiVersion` as a property. Hence, I need to do these changes to get the `group`.

[line](https://github.com/openshift/console/pull/4352/files#diff-bc19a445cd55d99705e404fae2c786edL71) basically checks whether the actions(from the `ClusterServiceVersionAction` extension) need to be added or not.

Reply:
As the checks have been added here only, hence I have also used the method to do so.

Reply:
You should rename the parameter and the type. It's no longer a k8s object. I'd suggest `model: K8sKind`

Why does the kind object not have an api version property? It should.

Reply:
Oh, are we somethings passing a k8s object and sometimes a model? That's not a good pattern. We should always pass the same type to this function and not attempt to do any detection.

Reply:
I see for having the CSV action enabled, we need to match kind and apigroup (apigroup because of [this](https://github.com/openshift/console/pull/2840#issue-321573215)). 
The `getActions(selectedObj)` is called once for Kebab actions and menu actions. 
For Kebab Actions what we are passing is a k8sResource  but for menuActions a string (that I think because getActions is expecting something in params).

While passing K8sKind object to `getActions` call, would work to show CSV actions, only if you drill down the right `apiversion` to this format "group/version". 

One approach i can think of:
```
groupVersionFor(apiVersionForModel(selectedObj)).group === action.properties.apiGroup
```

@gnehapk @spadgett ^^

Reply:
We should always pass the model to this method. We shouldn't sometimes pass the model and sometimes pass an instance. We have the model in both contexts already.

Reply:
@spadgett thanks for the inputs,  any pointers on the models for kebabactions ? The `OperandTableRowProps` seems to provide only a [k8sresource](https://github.com/openshift/console/blob/f4b7d0f643f62d099cc9ae7d11018be913435371/frontend/packages/operator-lifecycle-manager/src/components/operand.tsx#L255-L279). 

Or one more thing we can do if you consider:

Just passing kind and apigroup in the `getActions` function only.
```
const getActions = ({kind, apiGroup}) => {
....
```
Thanks!

Reply:
I think the simplest thing to do is to change `getActions` to take a referene

```ts
const getActions = (reference: K8sResourceKindReference) => {
```

and pass it

```tsx
        <ResourceKebab actions={getActions(referenceFor(obj)} kind={referenceFor(obj)} resource={obj} />
```

and

```ts
    menuActions={getActions(props.modelRef)}
```

Reply:
Indeed :+1: